lib/logger/lib/logger/watcher.ex

```diff
@@ -28,18 +26,24 @@ defmodule Logger.Watcher do
   Removes the given handler.
   """
   def unwatch(mod, handler) do
-    GenEvent.remove_handler(mod, handler, :ok)
+    result = :gen_event.delete_handler(mod, handler, :ok)
+    _ = Supervisor.terminate_child(@name, {__MODULE__, {mod, handler}})
+    result
```

Comment:
Ah, ok, I understand. I tried switching it to just `terminate_child`, but because it doesn't delete the child spec, calls to remove a previous-added-and-removed backend do not produce the `:not_found` error. I changed it so that the body of `unwatch` is now:

```elixir
child_id = {__MODULE__, {mod, handler}}
Supervisor.terminate_child(@name, child_id)
Supervisor.delete_child(@name, child_id)
```

Seems like that is OK to me, but just wanted to check that I'm not missing a new issue that causes. :)